pymc/aesaraf.py

```diff
@@ -537,23 +537,64 @@ def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
     xs: List[TensorVariable],
     vars: List[TensorVariable],
-    shared,
+    shared: Dict[TensorVariable, TensorSharedVariable],
     make_shared: bool = False,
-):
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Takes a list of TensorVariables and joins their non shared inputs into a single input.
 
     Parameters
     ----------
-    point: a sample point
-    xs: list of Aesara tensors
-    vars: list of variables to join
+    point: a sample point associated with a model and variable inputs
+    xs: list of TensorVariable to replace subgraphs with vars and shared
+    vars: list of TensorVariable to reshape, join, and return as inarray
+    shared: dict of TensorVariable and their associated TensorSharedVariable in
+        subgraph replacement
+    make_shared: bool flag to use point argument to build subgraphs of xs_special
 
     Returns
     -------
-    tensors, inarray
-    tensors: list of same tensors but with inarray as input
-    inarray: vector of inputs
+    xs_special, inarray
+    xs_special: list of same xs TensorVariables but with inarray and shared in subgraphs
+    inarray: TensorVariable built from vars argument
+
+    Examples
+    --------
```

Comment:
I suggest a pure Aesara example first:

```suggestion
    --------
    Join the inputs of a simple Aesara graph
    
    .. code-block:: python
    
        import aesara.tensor as at
        import numpy as np

        from pymc.aesaraf import join_nonshared_inputs

        # Original non-shared inputs
        x = at.scalar("x")
        y = at.vector("y")
        # Original output
        out = x + y
        print(out.eval({x: np.array(1), y:np.array([1, 2, 3])}))  # [2, 3, 4]

        # New output and shared input
        [new_out], joined_inputs = join_nonshared_inputs(
          point={"x": np.zeros(()), "y": np.zeros(3)},  # Only shapes matter
          xs=[out],
          vars=[x, y],
          shared=[],
        )

        print(new_out.eval({joined_inputs: np.array([1, 1, 2, 3])}))  # [2, 3, 4]
        
```
```